tests/ImageSharp.Tests/TestUtilities/TestImageExtensions.cs

```diff
@@ -22,6 +22,38 @@ namespace SixLabors.ImageSharp.Tests
 
     public static class TestImageExtensions
     {
+        /// <summary>
+        /// TODO: This should be a common processing method! The image.Opacity(val) multiplies the alpha channel!
+        /// </summary>
+        /// <typeparam name="TPixel"></typeparam>
+        /// <param name="ctx"></param>
+        public static void MakeOpaque<TPixel>(this IImageProcessingContext<TPixel> ctx)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            ctx.Apply(
+                img =>
+                    {
+                        using (var temp = new Buffer2D<Vector4>(img.Width, img.Height))
+                        {
+                            Span<Vector4> tempSpan = temp.Span;
+                            foreach (ImageFrame<TPixel> frame in img.Frames)
+                            {
+                                Span<TPixel> pixelSpan = frame.GetPixelSpan();
+
+                                PixelOperations<TPixel>.Instance.ToVector4(pixelSpan, tempSpan, pixelSpan.Length);
+
+                                for (int i = 0; i < tempSpan.Length; i++)
+                                {
+                                    ref Vector4 v = ref tempSpan[i];
+                                    v.W = 1.0f;
```

Comment:
This only works if your vector range is `0-1`, for  `Short4` ranges from `-37267` to `37267`. That's why I've relied on the image format to handle transparency so far.

Reply:
I do wonder whether we should be using the ScaledVector approach for all our methods. it would slow things down slightly for the more exotic types though.

Reply:
Yeah that ranging stuff is quite complicated, I still can't follow it. Maybe we can have `ToVector4` and `ToScaledVector4` side by side, letting the ImageProcessor to pick one of them. The refactor path could be also easier.
Neverheless this is just a basic implementation for testing purposes used by `***IsNotBoundToSinglePixelType` theories, and we are testing just a few pixel types in those.

Reply:
Let's go for it then! 
(Post 1.0 I guess)